src/main/java/com/minecolonies/coremod/util/BarbarianUtils.java

```diff
@@ -0,0 +1,297 @@
+        final List<EntityLivingBase> entityList = CompatibilityUtils.getWorld(entity).getEntitiesWithinAABB(
+          EntityLivingBase.class,
+          entity.getEntityBoundingBox().expand(
+            distanceFromEntity,
+            3.0D,
+            distanceFromEntity),
+          Entity::isEntityAlive);
+
+        return entityList.stream().filter(BarbarianUtils::isBarbarian);
+    }
+
+    /**
+     * Enacts the Raid Event.
+     *
+     * @param colony The colony for which the raid is occurring in.
+     */
+    public static void EventRaid(final Colony colony)
+    {
+        final World world = colony.getWorld();
+
+        if (world == null)
+        {
+            return;
+        }
+
+        final boolean isDay = world.isDaytime();
+
+        if (isDay && colony.getHasRaided())
+        {
+            colony.setRaidWillHappen(raidThisNight(world));
+            colony.setHasRaided(false);
+        }
+        else if (isDay && !colony.getHasRaided())
+        {
+            colony.setHasRaided(false);
+        }
+        else if (!isDay && !colony.getHasRaided() && colony.getWillRaid() && Configurations.doBarbariansSpawn && (world.getDifficulty() != EnumDifficulty.PEACEFUL))
+        {
+            System.out.println("Did the thing");
+            NumberOfSpawns(colony);
+
+            final BlockPos targetSpawnPoint = calculateSpawnLocation(world, colony);
+
+            if (Configurations.enableInDevelopmentFeatures)
+            {
+                LanguageHandler.sendPlayersMessage(
+                  colony.getMessageEntityPlayers(),
+                  "Horde Spawn Point: " + targetSpawnPoint);
+            }
+
+            LanguageHandler.sendPlayersMessage(
+              colony.getMessageEntityPlayers(),
+              "event.minecolonies.raidMessage");
+
+            BarbarianSpawnUtils.spawn(barbarian, numberOfBarbarians, targetSpawnPoint, world);
+            BarbarianSpawnUtils.spawn(archer, numberOfArchers, targetSpawnPoint, world);
+            BarbarianSpawnUtils.spawn(chief, numberOfChiefs, targetSpawnPoint, world);
+
+            colony.setHasRaided(true);
+        }
+        else if (!isDay && !colony.getHasRaided() && colony.getWillRaid() && Configurations.doBarbariansSpawn && (world.getDifficulty() == EnumDifficulty.PEACEFUL))
+        {
+            if (Configurations.enableInDevelopmentFeatures)
+            {
+                LanguageHandler.sendPlayersMessage(
+                  colony.getMessageEntityPlayers(),
+                  "(This is for asher) Asherslab: You are an idiot, it's still set on peaceful, love Asher");
+            }
+            colony.setHasRaided(true);
+        }
+        else if (!isDay && !colony.getHasRaided())
+        {
+            colony.setHasRaided(true);
+        }
+    }
+
+    /**
+     * Sets the number of spawns for each barbarian type
+     *
+     * @param colony The colony to get the RaidLevel from
+     */
+    private static void NumberOfSpawns(final Colony colony)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00100)